test/functional/autocmd/signal_spec.lua

```diff
@@ -8,39 +8,88 @@ local fn = n.fn
 local next_msg = n.next_msg
 local is_os = t.is_os
 local skip = t.skip
+local read_file = t.read_file
+local feed = n.feed
+local retry = t.retry
 
 if skip(is_os('win'), 'Only applies to POSIX systems') then
   return
 end
 
-local function posix_kill(signame, pid)
-  os.execute('kill -s ' .. signame .. ' -- ' .. pid .. ' >/dev/null')
-end
+describe("'autowriteall' on signal exit", function()
+  before_each(clear)
+
+  local function test_deadly_sig(signame, awa, should_write)
+    local testfile = 'Xtest_SIG' .. signame .. (awa and '_awa' or '_noawa')
+    local teststr = 'Testaaaaaaa'
```

Comment:
can do this to ensure the `testfile` is deleted after the test is finished (this is a feature of the "busted" test framework):



```suggestion
    local teststr = 'Testaaaaaaa'
    finally(function()
      os.remove(testfile)
    end)
```